### PR TITLE
Automatic text positioning

### PR DESCRIPTION
**Pull request by MFlisar**
I added following setters:

* enableAutoTextPosition
* autoTextPadding

And a basic very simply calculation that calculates if there is more space above the focus view or below and then sets the `TextView` padding accordingly. The gravity parameter will then be used in the remaining space automatically

Should solve https://github.com/faruktoptas/FancyShowCaseView/issues/20 in most cases. Can still be improved to check left/right as well

### CODE REVIEW

**Comment by faruktoptas** [comment]
I will check this feature asap.

File: library/src/main/java/me/toptas/fancyshowcase/Calculator.java [hunk 1]
```diff
@@ -81,6 +85,33 @@ public void setCirclePosition(int positionX, int positionY, int radius) {
         mHasFocus = true;
     }
 
+    public void calcAutoTextPosition(final View view, final int padding) {
+        view.post(new Runnable() {
```
Comment (faruktoptas) [inline library/src/main/java/me/toptas/fancyshowcase/Calculator.java]: Why did you use `view.post()` here?

File: library/src/main/java/me/toptas/fancyshowcase/Calculator.java [hunk 2]
```diff
@@ -81,6 +85,33 @@ public void setCirclePosition(int positionX, int positionY, int radius) {
...
+            @Override
+            public void run() {
+
+                // calc target view position
+                float top = roundRectTop(0, 0);
```
Comment (faruktoptas) [inline library/src/main/java/me/toptas/fancyshowcase/Calculator.java]: You should add `mFocusCircleRadiusFactor` to the calculation. This is the coefficient value for circle radius. 
I tried this:
```
new FancyShowCaseView.Builder(this)
                .focusOn(view)
                .focusCircleRadiusFactor(4)
                .title("Focus on View with larger circle")
                .build()
                .show();

```
And here is the result:

device-2017-05-28-110655

**Comment by faruktoptas** [comment]
@MFlisar are you still working on this issue?

**Comment by MFlisar** [comment]
No, my solution was always working fine.

To answer your question about the `view.post`: this way I'm sure the view is layed out and I can get it's parent and check it's width and height.

I've never used circles yet and for others my solution works quite well...

**Comment by faruktoptas** [comment]
Your solution works fine if `focusCircleRadiusFactor` is 1. For bigger values it calculates wrong position as shown in the screenshot. 

I create `Calculator` instance after view is laid out. I fixed it. `view.post` is not needed now.

So If you are not working on this issue I will close it.

**Comment by MFlisar** [comment]
I'm not working on it anymore. Back when I made this change the `view.post` was definitely necessary though...

You can close it if you want

**Comment by faruktoptas** [comment]
Thank you @MFlisar I will fix the focusCircleRadiusFactor problem then merge it.

**Comment by XiiaoKai** [comment]
Is this function added in already?

**Comment by XiiaoKai** [comment]
@faruktoptas Hope it can release on this week, if not we might need to switch library.

**Comment by faruktoptas** [comment]
Reimplemented with https://github.com/faruktoptas/FancyShowCaseView/commit/747ae80f8754065f33858b8c2d5dd1392f075ddb

**Pull request closed without merging**